addons/website/models/website.py

```diff
@@ -154,6 +154,19 @@ def write(self, values):
             self.env['ir.qweb'].clear_caches()
         return result
 
+    @api.multi
+    def unlink(self):
+        # Do not delete invoices, delete what's strictly necessary
+        attachments_to_unlink = self.env['ir.attachment'].search([
+            ('website_id', 'in', self.ids),
+            '|', '|',
+            ('key', '!=', False),  # theme attachment
+            ('url', 'ilike', '.custom.'),  # customized theme attachment
+            ('url', 'ilike', '.assets_'),
```

Comment:
.assets\_

```python
In [0]: x = 'website.assetssansunderscore'                                                                                                                                                                        

In [1]: env['ir.attachment'].create({'name': x, 'url':x})                                                                                                                                                         
Out[1]: ir.attachment(538,)

In [2]: env['ir.attachment'].search([('url', 'ilike', '.assets_')], order='id desc', limit=1).mapped('url')                                                                                                       
Out[2]: ['website.assetssansunderscore']

In [3]: env['ir.attachment'].search([('url', 'ilike', '.assets\_')], order='id desc', limit=1).mapped('url')                                                                                                      
Out[3]: ['/web/content/535-9a90afe/website.assets_editor.js']

```

+ indent the '|'

Reply:
Good catch, indeed it was not doing exactly what I wanted.
Note that I wrote it `.assets\\_` for linter escaping error (not sure there is a difference aside linter error between one or two `\` in python string, while on other cases their might be [1](https://stackoverflow.com/questions/38084864/how-to-escape-underscores-in-postgresql#comment70198430_38084865) [2](https://github.com/mysqljs/mysql/issues/650) ).

About the indent of `|`, do you prefer it written in 2 lines instead of one? Not sure if that's what u meant.